packages/multichain-network-controller/src/utils.ts

```diff
@@ -130,3 +159,203 @@ export const toMultichainNetworkConfigurationsByChainId = (
+  } catch (error) {
+    if (error instanceof Error) {
+      if (error.name === 'AbortError') {
+        throw new Error('Request timeout: Failed to fetch active networks');
+      }
+      throw error;
+    }
+
+    throw new Error(`Failed to fetch active networks: ${String(error)}`);
+  }
+}
+
+/**
+ * Parses a network string in the format "namespace:chainId:address".
+ *
+ * @param network - The network string to parse
+ * @returns The parsed components or null if invalid
+ */
+export function parseNetworkString(
+  network: string,
+): NetworkStringComponents | null {
+  const [namespace, chainId, address] = network.split(':');
+
+  if (!namespace || !address) {
+    return null;
+  }
+
+  // Validate address format based on namespace
+  switch (namespace as KnownCaipNamespace) {
+    case KnownCaipNamespace.Eip155:
+      if (!address.startsWith('0x') || !/^0x[0-9a-fA-F]{40}$/u.test(address)) {
+        return null;
+      }
+      break;
+    case KnownCaipNamespace.Solana:
+      if (!isSolanaAddress(address)) {
+        return null;
+      }
+      break;
+    case KnownCaipNamespace.Bip122:
+      // Bitcoin addresses can be legacy, segwit, or native segwit
+      // Need a utility function to validate bitcoin addresses
+      if (!/^(1|3|bc1)[a-zA-Z0-9]{25,62}$/u.test(address)) {
+        return null;
+      }
+      break;
+    default:
+      return null;
+  }
+
+  return {
+    namespace: namespace as KnownCaipNamespace,
+    chainId: chainId || '',
+    address: address as Hex,
+  };
+}
+
+/**
+ * Formats the API response into our state structure.
+ * Example input: ["eip155:1:0x123...", "eip155:137:0x123...", "solana:1:0xabc..."]
+ *
+ * @param response - The raw API response
+ * @returns Formatted networks by address
+ */
+export function formatNetworkActivityResponse(
+  response: ActiveNetworksResponse,
+): ActiveNetworksByAddress {
+  const networksByAddress: ActiveNetworksByAddress = {};
+
+  response.activeNetworks.forEach((network) => {
+    const components = parseNetworkString(network);
+    if (!components) {
+      return;
+    }
+
+    const { namespace, chainId, address } = components;
+
+    if (!networksByAddress[address]) {
+      networksByAddress[address] = {
+        namespace,
+        activeChains: [],
+      };
+    }
+
+    // Only add chainId to activeChains for EVM networks (non-EVM networks are not supported yet)
+    if (namespace === KnownCaipNamespace.Eip155 && chainId) {
+      networksByAddress[address].activeChains.push(chainId);
+    }
+  });
+
+  return networksByAddress;
+}
+
+/**
+ * Formats an account address with its corresponding CAIP prefix. Used to format the account IDs for the active networks API.
+ *
+ * @param address - The account address
+ * @param chainType - The type of chain (EVM, BTC, or SOLANA)
+ * @returns The formatted CAIP-10 account identifier
+ */
+export function formatCaipAccountId(
+  address: CaipAccountAddress,
+  chainType: ChainType,
+): string {
```

Comment:
Is it worth using [`toCaipAccountId` from `@metamask/utils`](https://github.com/MetaMask/utils/blob/7c9d3a22d69b0def075c1f630e0249cfae9ddfb6/src/caip-types.ts#L414) to ensure that the return type is `CaipAccountId`?